core/rawdb/accessors_chain.go

```diff
@@ -1749,3 +1820,84 @@ func DeleteUtxoToBlockHeight(db ethdb.KeyValueWriter, txHash common.Hash, index
 		db.Logger().WithField("err", err).Fatal("Failed to delete utxo to block height")
 	}
 }
+
+func ReadCoinbaseLockup(db ethdb.KeyValueReader, batch ethdb.Batch, ownerContract common.Address, beneficiaryMiner common.Address, lockupByte byte, epoch uint32) (*big.Int, uint32, uint16, common.Address) {
```

Comment:
you should write test cases to validate each of these cases, should be straight forward but will help us to prevent edge cases and bugs in general

Reply:
Make an issue and put it in my todo

Reply:
https://github.com/dominant-strategies/go-quai/issues/2462